### PR TITLE
Fix cross version FIPS compatibility tests

### PR DESCRIPTION
**Pull request by paulidale**
- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
Only the FIPS version commit to go to 3.1 and 3.0.

File: .github/workflows/provider-compatibility.yml [hunk 1]
```diff
@@ -10,9 +10,7 @@
 
 name: Provider compatibility across versions
 
-on: #[pull_request]
-  schedule:
-    - cron: '0 15 * * *'
+on: [pull_request, push]
```
Comment (t8m) [inline .github/workflows/provider-compatibility.yml:13]: I think running it on pull request is too much.

**Review by t8m — APPROVED** [review]
Approving the fix (not the change of triggering the run).

OK with urgent.

File: .github/workflows/provider-compatibility.yml (hunk 1 above)
Comment (paulidale) [reply .github/workflows/provider-compatibility.yml:13]: There is no other way to test it.

I think it should remain daily.

Comment (t8m) [reply .github/workflows/provider-compatibility.yml:13]: Yeah, then please add fixup commit reverting this back to scheduled

**Comment by t8m** [comment]
> Plus tests will fail until the version check is back-ported.

@paulidale I am confused, why the evp tests still fail even with the annotation added?

**Comment by paulidale** [comment]
I'm confused by it too.  I spent way too long trying to figure it out yesterday without success.

Locally, they pass in the CIs here they fail.

**Comment by paulidale** [comment]
Finally figured out the testing problem.

This test is designed to be run daily, thus there is no associated PR.  It's grabbing the `master` branch from git _directly_ and this does _not_ include the changes to the test cases that are present.  Hence the tests that require them fail.

This should be looked into somehow, but not here.

**Review by t8m — APPROVED** [review]
OK with urgent

**Review by mattcaswell — APPROVED** [review]
Ok with urgent

**Comment by t8m** [comment]
Merged to master branch. Thank you.

I do not think there is anything to backport here to stable branches as the test is scheduled -> master only and the FIPSversion checks are for testcases which are also only on master branch.

**Pull request closed without merging**